assignments/this.js

```diff
@@ -1,26 +1,62 @@
 /* The for principles of "this";
 * in your own words. explain the four principle for the "this" keyword below.
 *
-* 1. 
-* 2. 
-* 3. 
-* 4. 
+* 1. window binding is when a function with "this" is called without (context?), it defaults to the window object 
+* 2. implicit binding is when you call a function, whatever is left of the function is what "".this" is referring to.
+* 3. new binding is when the "new" keyword makes "this" refer to the new object being created...   ???
+* 4. explicit binding is when you use".call()", ".apply()", and ".bind()" to assign what you want "this" to refer to.
 *
 * write out a code example of each explanation above
 */
 
-// Principle 1
-
-// code example for Window Binding
-
-// Principle 2
-
-// code example for Implicit Binding
-
-// Principle 3
-
-// code example for New Binding
-
-// Principle 4
-
-// code example for Explicit Binding
\ No newline at end of file
+//                                      Principle 1
+//                               code example for Window Binding (done)
+
+// const sayAge = function() {
+//     console.log(this.age);
+// }
+// const me = {
+//     age: 23
+// }
+// sayAge(); // logs "undefined"
+
+//                                      Principle 2
+//                              code example for Implicit Binding (done)
+
+// const me = {
+//     name: 'Clark',
+//     age: 23,
+//     sayName: function(){
+//         console.log(this.name);
```

Comment:
```suggestion
//         console.log(this.name);
console.log(this)
```

Reply:
console.log(this) on each example see what it's referencing. It will help with understanding of the .this binding